Tests/ApolloTests/Cache/ReadWriteFromStoreTests.swift

```diff
@@ -609,4 +609,213 @@ class ReadWriteFromStoreTests: XCTestCase, CacheDependentTesting, StoreLoading {
       }
     }
   }
+
+  func testRemoveObjectByPatternCaseInsensitive() throws {
```

Comment:
This hasn't been the pattern used in the past, but I'd like to propose an improvement for naming tests functions. This describes vaguely what is being tested, but a more complete test name would be in the pattern:

"test_${thing-being-tested}_given${parameters-or-details}_(will|should|does|doesNot)${expected-behavior}"

So this would become `func test_removeObjectMatchingPattern_givenPatternNotMatchingCase_deletesCaseInsensitiveMatchingRecords()`

We don't need to go back and change all the test names now, but as we edit them or add new tests, I'd like us to follow this pattern. It makes for a much better understanding of what's being tested and what the actual expected behavior is. With the test above, I don't know if we **want** it to remove the record when it's case insensitive or **not** remove the record.

Reply:
@calvincestari @designatednerd Thoughts?

Reply:
I'm fine with this general pattern but if we're gonna do this we should go through and update the rest of the tests soon. 

I would say that regardless of whether we go with that pattern or not right now, @AnthonyMDev is correct that the test name is a little vague. 

Channel your Obj-C experience when naming tests and go stupidly specific (ex, the `testUploadingSingleFileInAnArrayWithAnotherFileForAnotherField` test I was showing y'all yesterday)

Reply:
I updated the test names in 9cbd23ab and created https://github.com/apollographql/apollo-ios/issues/1849 for us to track updating test names across all tests; we can pick at it slowly.